### ISSUE TITLE
[button] NVDA doesn't announce loading state example correctly

### ISSUE THREAD

**Issue by grace-snow** [mui/base-ui#4249]
# Bug report

## Current behavior

The "Loading states" variant includes a button that becomes temporarily disabled and changes its visible label from "Submit" to "Submitting". NVDA does not announce the "Submitting" text, only "Unavailable".

VoiceOver announces both the changed label and dimmed (disabled) state correctly.

## Expected behavior

All screen readers should announce "submitting".

## Reproducible example

1. On a Windows machine, turn on NVDA screen reader
2. Navigate to [Button: Loading states](https://base-ui.com/react/components/button#loading-states)
3. Activate the button
4. Note only "unavailable" is announced and not the visible "submitting" text

## Base UI version

v1.2.0
## Which browser are you using?

Chrome

## Which OS are you using?

Windows

## Which assistive tech are you using (if applicable)?

NVDA

## Additional context

The button could have an `aria-live` attribute on it by default.

**Comment by aarongarciah** [comment]
Thanks @grace-snow!
